source/server/active_stream_listener_base.h

```diff
@@ -135,5 +135,61 @@ class ActiveStreamListenerBase : public ActiveListenerImplBase,
   Event::Dispatcher& dispatcher_;
 };
 
+template <typename ActiveConnectionType>
```

Comment:
Please add class level comment explaining the role of this template class.  It is hard to understand how this fits into the bigger picture.

Reply:
+1 to avoid templates. I agree with @antoniovicente that it's hard to understand. Almost any template usage like this can be switched to a base class with virtual overrides where needed. Can we do that here?

/wait

Reply:
My current attempt is to add the comment.

I am exploring what I can do next: The base class is landed as the previous PR.
At the first glance I should be able to use that class but I need to merge the parent internal listener PR to verify.

/wait

Reply:
Almost confirmed that internal connection (single and collection) is not necessary. The template can be removed.
/wait

Reply:
Template removed!